File: leetcode104.py

class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    def maxDepth(self, root: TreeNode) -> int:
        if root == None:
            return 0
        return self.depth(root, 0)
        
    def depth(self, node:TreeNode, h:int):
        if node == None:
            return h
        h = h+1
        if node.left == None and node.right == None:
            return h
        h1 = self.depth(node.left, h)
        h2 = self.depth(node.right, h)
        return max(h1,h2)
        

File: leetcode104.go
// +build ignore

package main

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func maxDepth(root *TreeNode) int {
	if root == nil {
		return 0
	}
	return depth(root, 0)
}
func depth(node *TreeNode, h int) int {
	if node == nil {
		return h
	}
	h++
	if node.Left == nil && node.Right == nil {
		return h
	}
	h1 := depth(node.Left, h)
	h2 := depth(node.Right, h)
	if h1 > h2 {
		return h1
	} else {
		return h2
	}
}
